x/storage/keeper/permission.go

```diff
@@ -82,6 +85,9 @@ func (k Keeper) VerifyObjectPermission(ctx sdk.Context, bucketInfo *types.Bucket
 	if ownerAcc.Equals(operator) {
 		return permtypes.EFFECT_ALLOW
 	}
+	if operator.Empty() {
```

Comment:
It's wrong. You should put this check to line 83.

BTW, why did your test pass? Please re-check your code.